server/auth.js

```diff
@@ -1,137 +1,118 @@
+          const header = req.headers.authorization;
+          if (!header || !header.indexOf('Bearer ') === 0) {
+            return { isValid: false };
+          }
+          const token = header.split(' ')[1];
+          const isApiRequest = decoded && decoded.payload && decoded.payload.iss === `https://${config('AUTH0_DOMAIN')}/`;
+          const isDashboardAdminRequest = decoded && decoded.payload && decoded.payload.iss === config('PUBLIC_WT_URL');
+
+          const getKeyAsync = promisify(jwtOptions.resourceServer.key);
+          const jwtVerifyAsync = promisify(jwt.verify);
+
+          if (isApiRequest) {
+            if (decoded.payload.gty && decoded.payload.gty !== 'client-credentials') {
+              return { isValid: false };
             }
 
-            return jwt.verify(token, key, jwtOptions.resourceServer.verifyOptions, (err) => {
-              if (err) {
-                return callback(Boom.unauthorized('Invalid token', 'Token'), null, null);
-              }
-
-              if (decoded.payload.gty && decoded.payload.gty !== 'client-credentials') {
-                return callback(Boom.unauthorized('Invalid token', 'Token'), null, null);
-              }
-
-              if (!decoded.payload.sub.endsWith('@clients')) {
-                return callback(Boom.unauthorized('Invalid token', 'Token'), null, null);
-              }
+            if (!decoded.payload.sub.endsWith('@clients')) {
+              return { isValid: false };
+            }
 
-              if (decoded.payload.scope && typeof decoded.payload.scope === 'string') {
-                decoded.payload.scope = decoded.payload.scope.split(' '); // eslint-disable-line no-param-reassign
-              }
+            const resourceServerKey = await getKeyAsync(decoded);
 
-              return callback(null, true, decoded.payload);
-            });
-          });
-        } else if (decoded && decoded.payload && decoded.payload.iss === config('PUBLIC_WT_URL')) {
-          return jwt.verify(
-            token,
-            jwtOptions.dashboardAdmin.key,
-            jwtOptions.dashboardAdmin.verifyOptions,
-            (err) => {
-              if (err) {
-                return callback(Boom.unauthorized('Invalid token', 'Token'), null, null);
-              }
+            if (!resourceServerKey) {
+              return { isValid: false };
+            }
 
-              if (!decoded.payload.access_token || !decoded.payload.access_token.length) {
-                return callback(Boom.unauthorized('Invalid token', 'Token'), null, null);
-              }
+            // this can throw if there is an error
+            await jwtVerifyAsync(token, resourceServerKey, jwtOptions.resourceServer.verifyOptions);
 
-              decoded.payload.scope = scopes.map(scope => scope.value);
-              return callback(null, true, decoded.payload);
+            return { credentials: decoded.payload, isValid: true };
+          }
+          if (isDashboardAdminRequest) {
```

Comment:
Should we be applying handling if a token is neither a dashboard or api request? Would this just not return a value right now?

Reply:
I don't think there is a path for it to **not** be a `isDashboardAdminRequest`. I kept the same logic that was already there to limit the changes. I also was asking about is there a possible way for `isApiRequest` path to be hit, and this app isn't configured to really work in that way. This JWT strategy is really for those admin routes

Reply:
Got this fixed